File: frontend/src/app/infrastructure/person-factory.ts
import { Person } from '../domain/person';
import { PersonPayload, PersonResponse } from './person-dto';

export function createPerson(personDto: PersonResponse): Person {
  return new Person(
    personDto.uuid,
    personDto.dni,
    personDto.firstName,
    personDto.lastName,
    personDto.sex,
    new Date(`${personDto.birthday}T05:00:00.000Z`),
    personDto.hobbies,
    personDto.address,
    personDto.phone,
    personDto.email
  );
}

export function exportPerson(person: Person): PersonPayload {
  return {
    dni: person.dni,
    firstName: person.firstName,
    lastName: person.lastName,
    address: person.address,
    sex: person.sex,
    birthday: person.birthday.toISOString(),
    phone: person.phone,
    email: person.email,
    hobbies: person.hobbies,
  };
}

File: backend/src/main/java/org/danielcastillo/backend/application/factories/PersonFactory.java
package org.danielcastillo.backend.application.factories;

import org.danielcastillo.backend.application.PersonDTO;
import org.danielcastillo.backend.infrastructure.PersonJpa;

import java.util.UUID;

public class PersonFactory {
    public static PersonJpa create(PersonDTO personDTO) {
        var personJpa = new PersonJpa();
        personJpa.uuid = UUID.randomUUID().toString();
        personJpa.dni = personDTO.dni;
        personJpa.firstName = personDTO.firstName;
        personJpa.lastName = personDTO.lastName;
        personJpa.address = personDTO.address;
        personJpa.phone = personDTO.phone;
        personJpa.email = personDTO.email;
        personJpa.hobbies = personDTO.hobbies;
        personJpa.sex = personDTO.sex;
        personJpa.birthday = personDTO.birthday;

        return personJpa;
    }
}
